### ISSUE TITLE
meta/discussion: what will go in a `models` module

### ISSUE THREAD

**Issue by NickleDave** [vocalpy/vocalpy#47]
To go beyond bioacoustics and to actually serve as a core package for acoustic communication researchers, we need a `models` module that contains reference implementations of *models* of animal acoustic communication and behavior.

This is a meta issue to keep track of what that module might look like. Please feel free to comment and add to the discussion.

My initial thoughts:
* scope: "scikit-learn-sized" models, mainly written with Python + array-oriented numpy -- neural network models can live in vak
* how to group? by research area or by model type/framework?
  * any schema will be imperfect
  * research area would be something like: vocal learning, cultural evolution, syntax, evolution
  * model type would be something like: 
    * vocal learning
      * reinforcement learning
        * Troyer
        * Seung 
      * imitation learning
        * De Freitas 
      * babbling
        * models that Sita cites in review 
    * cultural evolution
      * agent-based models
        * Youngblood
        * Creanza 
    * syntax
      * context-free grammars, etc.
        * Sainburg
        * Beckers 
      * probabilistic-suffix tree
        * Markowtiz, canary 
      * markov
        *  Jin, Bengalese finch
    * evolution
      * game theory
      * Matt Spike models

:thinking: Maybe there's not enough overlap and it's ok to group by research area

**Comment by yardencsGitHub** [comment]
Seems to me that there are two labels: Types of models (e.g. graphical probabilistic models, mechanistic models ..), and usage (syntax, development .. ). It's not a 1-to-1 mapping.

**Comment by NickleDave** [comment]
Thanks @yardencsGitHub, agreed. We probably want something more like tags then a hierarchical structure. So we might structure the code base one way (say, math framework of model) but then provide multiple ways to view this in the docs (either by type of model or usage)

**Comment by NickleDave** [comment]
Nice to have would be bare-bones models like first order Markov + and descriptive stats from NLP like n-grams
